source/dub/dub.d

```diff
@@ -1666,10 +1666,13 @@ private class DependencyVersionResolver : DependencyResolver!(Dependency, Depend
 					logDiagnostic("Sub package %s doesn't exist in %s %s.", name, basename, dep.version_);
 					return null;
 				}
-			} else if (auto ret = m_dub.m_packageManager.getBestPackage(name, dep)) {
-				return ret;
 			} else {
-				logDiagnostic("External sub package %s %s not found.", name, dep.version_);
+				auto ret = dep.visit!(
+					(VersionRange rng) => m_dub.m_packageManager.getBestPackage(name, rng),
+					(any) => assert(0, "Cannot call getPackageRaw with a non-version dependency on subpackages"),
```

Comment:
it looks like it'd be possible that this `assert(false)` branch is called from

getSpecificConfigs -> getPackage -> getPackageRaw

Reply:
It shouldn't be, because otherwise the `assert(0)` in `Dependency.version_` would get triggered.
But I'll look at the CI failure.

Reply:
the dependency resolution logic is tricky to understand, so I don't really know where to start searching for such a case, but in general these non-version cases are not that well tested, so an edge case is unlikely to trigger in CI here